cmd/serve.go

```diff
@@ -19,13 +19,14 @@ import (
 	"os"
 	"time"
 
+	"github.com/sirupsen/logrus"
```

Comment:
@yogev-lb it's not clear what is this change and how it's related to commit message

Reply:
before the change this is the error you would have got:
```bash
[root@rack02-server79-vm05 dms]# docker logs discovery-client                                                                                                                                              
discovery-client configuration: model.AppConfig{Cores:[]int{0}, Logging:logging.Config{Filename:"/var/log/discovery-client.log", MaxAge:345600000000000, MaxSize:100, ReportCaller:true, Level:"info"}, Cli
entConfigDir:"/etc/discovery-client/discovery.d/", Debug:model.DebugInfo{Endpoint:"[::]:6060", Enablepprof:true, Metrics:true}, ReconnectInterval:5000000000, InternalDir:"/etc/discovery-client/internal/"
, LogPagePaginationEnabled:false, MaxIOQueues:0, AutoDetectEntries:model.AutoDetectEntries{Enabled:true, Filename:"detected-io-controllers", DiscoveryServicePort:0x1f49}, NvmeHostIDPath:"/tmp/discovery-c
lient/hostid", Kato:10}                                                                                                                                                                                    
level=info msg="******************** /discovery-client started ********************" func=cmd.serveCmdFunc file="serve.go:103"                                                                             
level=error msg="file /dev/nvme-fabrics does not exists. you may need to load nvme-tcp by running 'modprobe -a nvme-tcp'" func="application.(*App).Start" file="app.go:81" error="stat /dev/nvme-fabrics: n
o such file or directory"                                                                                                                                                                                  
level=error msg="failed to start application" func=cmd.serveCmdFunc file="serve.go:115" error="stat /dev/nvme-fabrics: no such file or directory"                                                          
Error: stat /dev/nvme-fabrics: no such file or directory                                                                                                                                                   
Usage:                                                                                                                                                                                                     
  discovery-client serve [flags]                                                                                                                                                                           
                                                                                                                                                                                                           
Flags:                                                                                                                                                                                                     
  -p, --autoDetectEntries.discoveryServicePort uint   discovery-service port (default 8009)                                                                                                                
  -e, --autoDetectEntries.enabled                     should we detect (default true)                                                                                                                      
  -f, --autoDetectEntries.filename string             name of the file we want to store this information (default "detected-io-controllers")                                                               
      --clientConfigDir string                        Directory to watch for discovery service configurations (default "/etc/discovery-client/discovery.d")                                                
      --debug.enablepprof                             Enable runtime profiling data via HTTP server. http://<endpoint>/debug/pprof/ (default true)                                                         
      --debug.endpoint string                         ip:port to expose debug and metric information (default "0.0.0.0:6060")                                                                              
      --debug.metrics                                 Expose prometheus metrics on http://<endpoint>/metrics (default true)                                                                                
  -h, --help                                          help for serve                                                                                                                                       
      --internalDir string                            Directory to store internal cache (default "/etc/discovery-client/internal")                                                                         
      --kato int                                      Host keep alive time out (default 10)                                                                                                                
      --logging.filename string                       filename to write log to                                                                                                                             
      --logging.level string                          Log level we support (default "debug")                                                                                                               
      --logging.maxSize int                           Maximum size in megabytes of the log file before it gets rotated. (defaults to 100MB). (default 100)                                                 
      --logging.maxage duration                       Time to wait until old logs are purged (default 96h0m0s)                                                                                             
      --logging.reportcaller                          Report func name and line number on log entry (default true)                                                                                         
      --maxIOQueues int                               Overrides the default number of I/O queues create by the driver. Zero value means no override (default driver value is number of cores).             
      --nvmeHostIDPath string                         file path containing nvme host id (default "/etc/nvme/hostid")                                                                                       
      --pollingInterval duration                      Polling interval for querying the discovery service. (default 5s)                                                                                    
                                                                                                                                                                                                           
Global Flags:                                                                                                                                                                                              
      --config string   config file (default is $HOME/.discovery-client/discovery-client.yaml) 
```

after the fix this is the error you would get:

```bash
[root@rack02-server79-vm05 dms]# docker logs discovery-client
discovery-client configuration: model.AppConfig{Cores:[]int{0}, Logging:logging.Config{Filename:"/var/log/discovery-client.log", MaxAge:345600000000000, MaxSize:100, ReportCaller:true, Level:"info"}, Cli
entConfigDir:"/etc/discovery-client/discovery.d/", Debug:model.DebugInfo{Endpoint:"[::]:6060", Enablepprof:true, Metrics:true}, ReconnectInterval:5000000000, InternalDir:"/etc/discovery-client/internal/"
, LogPagePaginationEnabled:false, MaxIOQueues:0, AutoDetectEntries:model.AutoDetectEntries{Enabled:true, Filename:"detected-io-controllers", DiscoveryServicePort:0x1f49}, NvmeHostIDPath:"/tmp/discovery-c
lient/hostid", Kato:10}
level=info msg="******************** /discovery-client started ********************" func=cmd.serveCmdFunc file="serve.go:105"
level=error msg="file /dev/nvme-fabrics does not exists. you may need to load nvme-tcp by running 'modprobe -a nvme-tcp'" func="application.(*App).Start" file="app.go:81" error="stat /dev/nvme-fabrics: n
o such file or directory"
level=error msg="failed to start application" func=cmd.serveCmdFunc file="serve.go:117" error="stat /dev/nvme-fabrics: no such file or directory"
Error: stat /dev/nvme-fabrics: no such file or directory
```

cobra by default will emit the help on any error, and in this context when running serve we don't need the help emitted, only the error